tests/test_parts.py

```diff
@@ -686,3 +686,76 @@ def test_bulk_create_parts_without_property_value(self):
             self.client._create_parts_bulk(
                 parts=[self.part_dict]
             )
+
+
+@skipIf(not TEST_FLAG_IS_PIM3, reason="This tests is designed for PIM version 3, expected to fail on older PIM3")
+@pytest.mark.skipif("os.getenv('TRAVIS', False) or os.getenv('GITHUB_ACTIONS', False)",
+                    reason="Skipping tests when using Travis or Github Actions, as not Auth can be provided")
+class TestBulkPartsDeletion(TestBetamax):
+    def setUp(self):
+        super().setUp()
+        self.parts = list()
+        self.wheel_model = self.project.model(name="Wheel")
+        self.bike_instance = self.project.part(name="Bike")
+
+        self.diameter_prop = self.wheel_model.property(name="Diameter")
+        self.spokes_prop = self.wheel_model.property(name="Spokes")
+        self.rim_material_prop = self.wheel_model.property(name="Rim Material")
+        self.tire_thickness_prop = self.wheel_model.property(name="Tire Thickness")
+
+        for idx in range(1, 5):
+            part_dict = {
+                "name": "Wheel {}".format(idx),
+                "parent_id": self.bike_instance.id,
+                "description": "",
+                "model_id": self.wheel_model.id,
+                "properties": [
+                    {
+                        "name": self.diameter_prop.name,
+                        "value": float("{}.{}".format(idx, idx)),
+                        "model_id": self.diameter_prop.id,
+                    },
+                    {
+                        "name": self.spokes_prop.name,
+                        "value": "{}".format(idx),
+                        "model_id": self.spokes_prop.id
+                    },
+                    {
+                        "name": self.rim_material_prop.name,
+                        "value": "Material {}".format(idx),
+                        "model_id": self.rim_material_prop.id
+                    },
+                    {
+                        "name": self.tire_thickness_prop.name,
+                        "value": float("{}.{}".format(idx, idx)),
+                        "model_id": self.tire_thickness_prop.id
+                    }
+                ]
+            }
+            self.parts.append(part_dict)
+        self.parts_created = list(self.client._create_parts_bulk(
+            parts=self.parts
+        ))
+
+    def tearDown(self):
```

Comment:
All tests are done in atomic state, so why would anything be remaining? And, if so, why not just `Part.objects.all().delete()`

Reply:
If the bulk_deletion fails for some reason then parts will still be left in the tree. I am interested in bringing the "Bike Project" to its original state and that means deleting the "Wheels" created in the setUp()